### PR TITLE
docs: illustrate example without expertsystem

### PR DESCRIPTION
**Pull request by redeboer**
Added a new page `usage/basics` that shows how to generate data for 1D and 2D expressions without using the `expertsystem`. This illustrates what happens behind the scenes in `tensorwaves`.

### CODE REVIEW

**Comment by redeboer** [comment]
@spflueger Have a look at the phrasing to see if you agree. I think this notebook also helps to rethink the design and naming (particularly in the `data` module).

File: docs/usage.ipynb [hunk 1]
```diff
@@ -11,7 +11,9 @@
...
+    "TensorWaves is a package for fitting general mathematical expressions to data distributions. The fundamentals behind the package are illustrated in {doc}`/usage/basics`.\n",
+    "\n",
+    "While general in design, the package is intended for doing {doc}`Partial Wave Analysis <pwa:index>`. First, the {mod}`expertsystem` determines which transitions are allowed from some initial state to a final state. It then formulates those transitions mathematically as an {ref}`amplitude model <usage:Construct a model>`. TensorWaves can then {meth}`~.Model.lambdify` this expression to some computational backend. Finally, TensorWaves {ref}`'fits' <usage:Optimize the model>` this model to some data sample. Optionally, this data sample can be {ref}`generated <usage:Generate data sample>` from the model.\n",
```
Comment (spflueger) [inline docs/usage.ipynb:16]: Optionally, **this** data sample can be -> ... **a** data sample ....

File: docs/usage/basics.ipynb [hunk 2]
```diff
@@ -0,0 +1,1040 @@
...
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The {doc}`/usage` page illustrates how to use {mod}`tensorwaves` in combination with the {mod}`expertsystem`. At core, however, {mod}`tensorwaves` is a package that can 'fit' arbitrary models to a data set using different backends, as well as generate toy data samples.\n",
+    "\n",
+    "These pages illustrate what's going on behind the scenes with some simple 1-dimensional and 2-dimensional models. Since we don't have a data sample to fit a model to, we follow the the same procedure as in the {doc}`/usage` page:\n",
+    "\n",
+    "1. Formulate a mathematical description with {mod}`sympy`.\n",
+    "2. Generate a 'toy' data sample with hit-and-miss.\n",
+    "3. Cast the model to some backend with {mod}`tensorwaves`.\n",
```
Comment (spflueger) [inline docs/usage/basics.ipynb:20]: Cast sounds a bit strange

File: docs/usage/basics.ipynb [hunk 3]
```diff
@@ -0,0 +1,1040 @@
...
+    "    expression=expression_1d,\n",
+    "    parameters=parameter_defaults,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Next, we use this {class}`.SympyModel` as a template to construct a {class}`.LambdifiedFunction` (abstract interface: {class}`.Function`). We choose {mod}`numpy` as computational backend, as we're not doing any optimizing yet (for optimizing it's better to use {obj}`jax <jax.jit>`)."
```
Comment (spflueger) [inline docs/usage/basics.ipynb:214]: abstract interface -> interface

File: docs/usage/basics.ipynb [hunk 4]
```diff
@@ -0,0 +1,1040 @@
...
+    "## Convert to backend"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "So far, all we did was using {mod}`sympy` to mathematically formulate a model. We now need to {func}`~sympy.utilities.lambdify.lambdify` that expression to some computational backend, so that we can efficiently generate data and/or optimize the parameters to 'fit' the model to some data sample.\n",
+    "\n",
+    "{mod}`tensorwaves` does this in two intermediate stages. First, the expression and parameter defaults are expressed in terms of a {class}`.SympyModel` (abstract interface: {class}`.Model`) that serves as a template for different computational backends."
```
Comment (spflueger) [inline docs/usage/basics.ipynb:195]: abstract interface -> interface

File: docs/usage/basics.ipynb [hunk 5]
```diff
@@ -0,0 +1,1040 @@
...
+    "draw_y_hit(x_random=x_value, y_random=0.17)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "There is one problem though: _how to determine $y_\\mathrm{max}$?_ In this example, we can just read off the value of $y_\\mathrm{max}$, or even compute it analytically from the original {class}`sympy.Expr <sympy.core.expr.Expr>`. This is not the case generally though, so we need to apply a trick.\n",
+    "\n",
+    "Since we are generating **uniformly distributed** random values values of $x$ and computing their $f(x)$ values, we can keep track of which values of $f(x)$ is the highest. Starting with $y_\\mathrm{max} = 0$ we just set $y_\\mathrm{max} = f(x)$ once $f(x) > y_\\mathrm{max}$ and _completely restart_ the generate loop. Eventually, some value of $x$ will lie near the absolute maximum of $f$ and the data generation will happily continue until the requested number of events has been reached."
```
Comment (spflueger) [inline docs/usage/basics.ipynb:438]: It should be noted that this procedure only works if the number of generated events is large. If the number of generated events is small the distribution generated by the this procedure will not accurately reflect the original function.

File: docs/usage/basics.ipynb [hunk 6]
```diff
@@ -0,0 +1,1040 @@
...
+   "metadata": {},
+   "source": [
+    "### Domain distribution"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "First of all, we need to randomly generate values of $x$. In this simple, 1-dimensional example, we could just use a random generator like {class}`numpy.random.Generator` feed its output to the {meth}`.LambdifiedFunction.__call__`=. Generally, though, we want to cover $n$-dimensional cases. So here's a small example function that generates a **uniform** distribution for each variable within a certain range:"
```
Comment (spflueger) [inline docs/usage/basics.ipynb:452]: {meth}`.LambdifiedFunction.__call__`= -> {meth}`.LambdifiedFunction.__call__`

**Review by spflueger — APPROVED** [review]
Nicely done. I have a few comments on some wordings and things that should be pointed out. Approved it already though

File: docs/usage/basics.ipynb (hunk 6 above)
Comment (redeboer) [reply docs/usage/basics.ipynb]: Well spotted

File: docs/usage/basics.ipynb (hunk 5 above)
Comment (redeboer) [reply docs/usage/basics.ipynb]: Added a warning admonition

File: docs/usage/basics.ipynb (hunk 2 above)
Comment (redeboer) [reply docs/usage/basics.ipynb]: Changed to Convert

**Comment by redeboer** [comment]
Thanks for the comments @spflueger, I addressed them in ca36e86

**Pull request merged**